### PR TITLE
rotateN checks mTargetPosition instead of current position

### PR DESCRIPTION
**Pull request by omentic**

### CODE REVIEW

**Comment by omentic** [comment]
@thehappycamden how is [this function](https://github.com/Spartronics4915/2020-InfiniteRecharge/blob/master/src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java#L160) supposed to work?

File: src/main/java/com/spartronics4915/frc2020/commands/IndexerCommands.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
     {
         public WaitForBallHeld(Indexer indexer)
         {
-            super(indexer::getIntakeBallLoaded);
+            super(indexer::getIntakeEmpty);
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/commands/IndexerCommands.java]: Why would we wait for the intake to be empty to stop the intake?

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 2]
```diff
@@ -124,9 +124,9 @@ public boolean getSlotBallLoaded()
      * <p>
      * The style of proximity sensor we use requires MANUAL calibration.
      *
-     * @return Whether a ball is held
+     * @return Whether a ball is not held
      */
-    public boolean getIntakeBallLoaded()
+    public boolean getIntakeEmpty()
     {
         return !mIntakeProxSensor.get();
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: This change doesn’t make any sense.

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 3]
```diff
@@ -247,6 +247,7 @@ public int getBalls()
     public boolean areFinsAligned()
     {
         double positionMod90 = mIndexerMotor.getEncoder().getPosition() % 90;
-        return (positionMod90 >= 85 || positionMod90 <= 5); // if in a safe space to load a ball
+        return (positionMod90 >= (90 - Constants.Indexer.Spinner.kPositionTolerance)
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: This is fixed in Camden’s branch.

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 4]
```diff
@@ -139,8 +139,8 @@ public void rotateN(double N)
     {
         if (N != 0)
         {
-            double deltaPosition = 0.25 * N; // Cast N to double and convert to rotations
-            mTargetPosition += deltaPosition;
+            double deltaPosition = 0.25 * N; // Convert N to rotations
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: This is fixed in Camden’s branch.

**Review by pietroglyph — CHANGES_REQUESTED** [review]

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 5]
```diff
@@ -115,7 +115,7 @@ public void setZero()
      */
     public boolean getSlotBallLoaded()
     {
-        return mOpticalProxSensor.get();
+        return !mOpticalProxSensor.get();
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Fixed in Camden's PR

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 6]
```diff
@@ -132,7 +132,7 @@ public void rotateN(double N)
         if (N != 0)
         {
             double deltaPosition = 0.25 * N; // Cast N to double and convert to rotations
-            mTargetPosition += deltaPosition;
+            mTargetPosition = deltaPosition + mIndexerMotor.getEncoder().getPosition();
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java:135]: This change is not desirable. We want to keep track of target positions. If we used this PR then the target position would drift.

**Review by pietroglyph — CHANGES_REQUESTED** [review]

**Pull request closed without merging**